### ISSUE TITLE
Enter/Return not working - flameshot gui --clipboard --path /home/elijah/Pictures/

### ISSUE THREAD

**Issue by ElijahLynn** [flameshot-org/flameshot#2355]
K, I got the <Return> key working but don't understand why. 

I changed:

`TYPE_COPY=Return`

to:

`TYPE_COPY=Ctrl+C`

And now <return> works, not Ctrl+C though, which is the opposite of what I was expecting. I'll open a new issue I suppose.

_Originally posted by @ElijahLynn in https://github.com/flameshot-org/flameshot/issues/2275#issuecomment-1028456897_

**Comment by mmahmoudian** [comment]
In the other thread you mentioned the following:

> I am on Arch using latest commit > 4291793.

From what I understand after reading all your posts, what you are reporting is that of you have set Return for some other action, the "Accept" will not work with Return anymore, which it should not be the case.

@veracioux since you implemented this feature relatively recently, can you check this out? I'm on the phone and cannot try to reproduce this.

**Comment by borgmanJeremy** [comment]
I think this is what is happening:

1. When you bind Return to TYPE_COPY, it is overriding the hard coded binding to "accept". 
2. When Flameshot is run in argument mode, there is no copy button, so Return does nothing
3. Since accept is not being triggered nothing happens.

When you change it to CTRL+C, Return is now properly bound to accept. 

Proposed solution:
If flameshot is run with arguments make sure Return is always bound to accept.

**Comment by FichteFoll** [comment]
Thanks for the explanation. I was able to get my return key working again in raw mode after I bound both copy and save to different keys.

**Comment by FichteFoll** [comment]
@mmahmoudian but as far as I can tell, that only controls whether the button is visible, not how I can invoke this functionality from the keyboard, which is how I prefer to confirm my selection as it's way easier to hit than one button out of 20.

**Comment by borgmanJeremy** [comment]
Turns out this is not a good first issue lol. 

## Background
The module that parses the config file and associated shortcut is 'standalone" in the sense that it does not tightly couple with the state of flameshot. This is generally a good thing because then our config file is parsed the same way no matter how flameshot is invoked. However, it means that we don't ignore "final action" hotkeys when running in CLI mode. 

Also, if a default assigned hotkey is ever used by a user, we omit the default hotkey: https://github.com/flameshot-org/flameshot/blob/4aebf2b568c397ec191530e6497825a6c64a0b59/src/utils/confighandler.cpp#L425

Since we have "Return" bound as the default hotkey for accept, when a user maps "Return" to another function the accept tool loses it's hotkey. This is a bit counterintuitive to most users because the accept tool is only present when running in CLI mode.

## Potential Solutions
### Solution 1
Hardcode the "return" key to "accept". We already do this with other keys like escape, but it means the Enter key will only be useful when running in CLI mode. 

### Solution 2
Leave the behavior as is but clean up the documentation to make this behavior more obvious. If you bind Enter to another function you should re-bind the accept tool to another hotkey.

### Solution 3
Detect if running in CLI mode and "drop" hotkeys used for final actions. This would solve this immediate bug, but has serious downsides in the way it impacts the config parser. Furthermore, if a user binds "Enter" to a non final action like the pencil tool, this bug will come back. I'm ruling out this option but kept it here because its the most obvious solution. 



Sorry for the long post. @mmahmoudian any thoughts between solutions 1 and 2? I'm leaning towards option 2 because it makes sense to me if you bind Return to another function you must rebind accept.

**Comment by mmahmoudian** [comment]
@borgmanJeremy Thanks for the detailed explanation. I think I more understand the whole issue.

As you said, the 3rd solution sounds to be the best from users' perspective, but it would make a hell of a spaghetti code if we should add so make exceptions. There can be a potential solution to this though, the config parser can cross-check the following for conflicting shortcuts:
- tools vs. final actions
- tools vs. Accept
And if there was no conflict (ignoring any conflicts between "final actions vs. Accept"), it returns the key bindings to the main process. Considering that the final actions and Accept are mutually exclusive, there will be no actual collision of shortcuts when the main process goes through the CLI mode route or not. This way:
1. We have parsed the config only once for each run of daemon
2. We don't need to implement any exceptions

If this alternative/amendment doesn't work or not possible to implement, I'll go with "solution 2" too

**Comment by FichteFoll** [comment]
If the accept and final action bindings must be the same key according to solution 2, couldn't it be reduced to just one binding used by both modes? That would remove the potential for user error and confusion.
